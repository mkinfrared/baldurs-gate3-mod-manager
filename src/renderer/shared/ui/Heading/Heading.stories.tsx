import { Meta, StoryObj } from "@storybook/react";

import { Heading } from "./Heading";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Heading",
  component: Heading,
  tags: ["autodocs"],
} satisfies Meta<typeof Heading>;

const Default: Story = {
  render: (args) => (
    <div className={css.Story}>
      <Heading {...args}>Example</Heading>
      <div className={css.variants}>
        <Heading variant="h1">Variants</Heading>
        <Heading variant="h1">Heading 1</Heading>
        <Heading variant="h2">Heading 2</Heading>
        <Heading variant="h3">Heading 3</Heading>
        <Heading variant="h4">Heading 4</Heading>
        <Heading variant="h5">Heading 5</Heading>
        <Heading variant="h6">Heading 6</Heading>
      </div>
    </div>
  ),
};

export { Default };

export default meta;
