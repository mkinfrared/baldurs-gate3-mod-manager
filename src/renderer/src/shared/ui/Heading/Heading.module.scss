.Heading {
  color: inherit;
  font-stretch: condensed;
  font-style: normal;
  font-weight: bold;
  letter-spacing: normal;
  line-height: normal;
  margin: 1rem 0;
}

h1.Heading {
  font-size: 3.4rem;
  line-height: 4.8rem;
}

h2.Heading {
  font-size: 2.2rem;
  line-height: 3.2rem;
}

h3.Heading {
  font-size: 1.8rem;
  line-height: 2.6rem;
}

h4.Heading {
  font-size: 1.4rem;
  line-height: 2rem;
}

h5.Heading {
  font-size: 1.2rem;
  line-height: 1.8rem;
}

h6.Heading {
  font-size: 1.4rem;
}
