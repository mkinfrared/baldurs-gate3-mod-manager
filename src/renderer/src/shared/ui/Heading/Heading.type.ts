export type HeadingProps = {
  className?: string;
  children?: React.ReactNode;
  onClick?: (event: React.MouseEvent<HTMLHeadingElement>) => void;
  variant?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
};
