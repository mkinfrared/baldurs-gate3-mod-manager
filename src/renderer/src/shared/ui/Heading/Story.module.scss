.Story {
  .variants {
    margin: 2rem 0;
  }
}
