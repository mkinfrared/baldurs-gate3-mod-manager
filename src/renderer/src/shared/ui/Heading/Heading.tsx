import { memo } from "react";

import { classNames } from "@renderer/shared/lib/helpers";

import css from "./Heading.module.scss";
import { HeadingProps } from "./Heading.type";

const Heading = ({ className, children, variant = "h3" }: HeadingProps) => {
  switch (variant) {
    case "h1":
      return (
        <h1
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h1>
      );
    case "h2":
      return (
        <h2
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h2>
      );
    case "h3":
      return (
        <h3
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h3>
      );
    case "h4":
      return (
        <h4
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h4>
      );
    case "h5":
      return (
        <h5
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h5>
      );
    default:
      return (
        <h6
          className={classNames(css.Heading, className)}
          data-testid="Heading"
        >
          {children}
        </h6>
      );
  }
};

export { Heading };

export default memo(Heading);
