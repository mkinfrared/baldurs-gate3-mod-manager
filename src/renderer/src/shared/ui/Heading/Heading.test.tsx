import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Heading } from "./Heading";

describe("<Heading />", () => {
  const testId = "Heading";
  const Component = <Heading />;

  it("should be defined", () => {
    expect(Heading).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should render an h3 tag by default when no variant prop is passed", () => {
    const { getByTestId } = render(Component);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H3");
  });

  it("should render an h1 tag when variant is h1", () => {
    const variant = "h1";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H1");
  });

  it("should render an h2 tag when variant is h2", () => {
    const variant = "h2";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H2");
  });

  it("should render an h3 tag when variant is h3", () => {
    const variant = "h3";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H3");
  });

  it("should render an h4 tag when variant is h4", () => {
    const variant = "h4";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H4");
  });

  it("should render an h5 tag when variant is h5", () => {
    const variant = "h5";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H5");
  });

  it("should render an h6 tag when variant is h6", () => {
    const variant = "h6";
    const { getByTestId } = render(<Heading variant={variant} />);
    const heading = getByTestId(testId);

    expect(heading.tagName).toBe("H6");
  });
});
