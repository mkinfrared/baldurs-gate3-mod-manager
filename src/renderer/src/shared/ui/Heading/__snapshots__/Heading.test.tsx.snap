// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Heading /> > should match the snapshot 1`] = `
<div>
  <h3
    class="Heading"
    data-testid="Heading"
  />
</div>
`;
